Marlin/Version.h

```diff
@@ -45,6 +45,16 @@
 #define STRING_DISTRIBUTION_DATE "2016-04-16 12:00"
 
 /**
+ * Required minimum Configuration.h and Configuration_adv.h file versions.
+ *
+ * You must increment this version number for every significant change such as,
+ * but not limited to: ADD, DELETE RENAME OR REPURPOSE any directive/option on
+ * the configuration files.
+ */
+#define REQUIRED_CONFIGURATION_H_VERSION 1
+#define REQUIRED_CONFIGURATION_ADV_H_VERSION 1
```

Comment:
I think we should only put version numbers the configurations in real release versions. This might be version 1, and the configs with 1.1.1 might be version 2 if they change very much. But in between they should just be considered works-in-progress with options that come and go as we figure things out.